src/plots/cartesian/axes.js

```diff
@@ -4523,3 +4574,38 @@ function setShiftVal(ax, axShifts) {
         axShifts[ax.overlaying][ax.side] :
         (ax.shift || 0);
 }
+
+/**
+ * Checks if the given period is at least the period described by the tickformat or larger. If that
+ * is the case, they are compatible, because then the tickformat can be used to describe the period.
+ * E.g. it doesn't make sense to put a year label on a period spanning only a month.
+ * @param {number} period in ms
+ * @param {string} tickformat
+ * @returns {boolean}
+ */
+function periodCompatibleWithTickformat(period, tickformat) {
+    if(/%f/.test(tickformat)) {
+        if(period < ONEMICROSEC) return false;
+    } else if(/%L/.test(tickformat)) {
+        if(period < 1) return false;
+    } else if(/%[SX]/.test(tickformat)) {
+        if(period < ONESEC) return false;
+    } else if(/%M/.test(tickformat)) {
+        if(period < ONEMIN) return false;
+    } else if(/%[HI]/.test(tickformat)) {
+        if(period < ONEHOUR) return false;
+    } else if(/%p/.test(tickformat)) {
+        if(period < HALFDAY) return false;
+    } else if(/%[Aadejuwx]/.test(tickformat)) {
+        if(period < ONEDAY) return false;
+    } else if(/%[UVW]/.test(tickformat)) {
+        if(period < ONEWEEK) return false;
+    } else if(/%[Bbm]/.test(tickformat)) {
+        if(period < ONEMINMONTH) return false;
+    } else if(/%[q]/.test(tickformat)) {
+        if(period < ONEMINQUARTER) return false;
+    } else if(/%[Yy]/.test(tickformat)) {
+        if(period < ONEMINYEAR) return false;
+    }
+    return true;
```

Comment:
You may consider refactoring this to something like
```js
function periodCompatibleWithTickformat(period, tickformat) {
    return !(
        (/%f/.test(tickformat) && period < ONEMICROSEC)  ||
        (/%L/.test(tickformat) && period < 1) ||
        (/%[SX]/.test(tickformat) && period < ONESEC) ||
        (/%M/.test(tickformat) && period < ONEMIN) ||
        (/%[HI]/.test(tickformat) && period < ONEHOUR) ||
        (/%p/.test(tickformat) && period < HALFDAY) ||
        (/%[Aadejuwx]/.test(tickformat) && period < ONEDAY) ||
        (/%[UVW]/.test(tickformat) && period < ONEWEEK) ||
        (/%[Bbm]/.test(tickformat) && period < ONEMINMONTH) ||
        (/%[q]/.test(tickformat) && period < ONEMINQUARTER) ||
        (/%[Yy]/.test(tickformat) && period < ONEMINYEAR)
    );
}
```

and then to perform faster tests to something like:
```js
function periodCompatibleWithTickformat(period, tickformat) {
    return !(
        (period < ONEMICROSEC && /%f/.test(tickformat))  ||
        (period < 1 && /%L/.test(tickformat)) ||
        (period < ONESEC && /%[SX]/.test(tickformat)) ||
        (period < ONEMIN && /%M/.test(tickformat)) ||
        (period < ONEHOUR && /%[HI]/.test(tickformat)) ||
        (period < HALFDAY && /%p/.test(tickformat)) ||
        (period < ONEDAY && /%[Aadejuwx]/.test(tickformat)) ||
        (period < ONEWEEK && /%[UVW]/.test(tickformat)) ||
        (period < ONEMINMONTH && /%[Bbm]/.test(tickformat)) ||
        (period < ONEMINQUARTER && /%[q]/.test(tickformat)) ||
        (period < ONEMINYEAR && /%[Yy]/.test(tickformat))
    );    
}
```

Reply:
Ah, I thought this'd be equivalent as well, however, the difference is that in my current code, as soon as a tickformat regex matches, the others aren't tested anymore.

E.g. if I have `tickformat = "%d, %Y"` and `period === ONEDAY` , it should return "compatible". It should not go on to test `(period < ONEMINYEAR && /%[Yy]/.test(tickformat)` as this would result in "not compatible".

I refactored the code to this to make it clearer.

```javascript
    if(/%f/.test(tickformat)) {
        return period >= ONEMICROSEC;
    } else if(/%L/.test(tickformat)) {
        return period >= 1;
    } else if(/%[SX]/.test(tickformat)) {
        return period >= ONESEC;
    } else if(/%M/.test(tickformat)) {
        return period >= ONEMIN;
    } else if(/%[HI]/.test(tickformat)) {
        return period >= ONEHOUR;
    } else if(/%p/.test(tickformat)) {
        return period >= HALFDAY;
    } else if(/%[Aadejuwx]/.test(tickformat)) {
        return period >= ONEDAY;
    } else if(/%[UVW]/.test(tickformat)) {
        return period >= ONEWEEK;
    } else if(/%[Bbm]/.test(tickformat)) {
        return period >= ONEMINMONTH;
    } else if(/%[q]/.test(tickformat)) {
        return period >= ONEMINQUARTER;
    } else if(/%[Yy]/.test(tickformat)) {
        return period >= ONEMINYEAR;
    }
    return true;
```

Reply:
Thanks for the note.
This looks way better now.
Just in case you want to drop several if else statements and combine the return calls the following may work:
```js
function periodCompatibleWithTickformat(period, tickformat) {
    return (
        /%f/.test(tickformat) ? period >= ONEMICROSEC :
        /%L/.test(tickformat) ? period >= 1 :
        /%[SX]/.test(tickformat) ? period >= ONESEC :
        /%M/.test(tickformat) ? period >= ONEMIN :
        /%[HI]/.test(tickformat) ? period >= ONEHOUR :
        /%p/.test(tickformat) ? period >= HALFDAY :
        /%[Aadejuwx]/.test(tickformat) ? period >= ONEDAY :
        /%[UVW]/.test(tickformat) ? period >= ONEWEEK :
        /%[Bbm]/.test(tickformat) ? period >= ONEMINMONTH :
        /%[q]/.test(tickformat) ? period >= ONEMINQUARTER :
        /%[Yy]/.test(tickformat) ? period >= ONEMINYEAR :
        true
    );
}
```

Reply:
Good idea, this is even clearer.